pkg/kubelet/preemption/preemption_test.go

```diff
@@ -90,82 +90,58 @@ func getTestCriticalPodAdmissionHandler(podProvider *fakePodProvider, podKiller
-		outputPods := podKiller.getKilledPods()
-		if !r.expectErr && outErr != nil {
-			t.Errorf("evictPodsToFreeRequests returned an unexpected error during the %s test.  Err: %v", r.testName, outErr)
-		} else if r.expectErr && outErr == nil {
-			t.Errorf("evictPodsToFreeRequests expected an error but returned a successful output=%v during the %s test.", outputPods, r.testName)
-		} else if !podListEqual(r.expectedOutput, outputPods) {
-			t.Errorf("evictPodsToFreeRequests expected %v but got %v during the %s test.", r.expectedOutput, outputPods, r.testName)
-		}
-		podKiller.clear()
-	}
-}
-
 func TestEvictPodsToFreeRequests(t *testing.T) {
 	type testRun struct {
 		testName              string
+		isPodKillerWithError  bool
 		inputPods             []*v1.Pod
 		insufficientResources admissionRequirementList
 		expectErr             bool
 		expectedOutput        []*v1.Pod
 	}
-	podProvider := newFakePodProvider()
-	podKiller := newFakePodKiller(false)
-	criticalPodAdmissionHandler := getTestCriticalPodAdmissionHandler(podProvider, podKiller)
 	allPods := getTestPods()
 	runs := []testRun{
 		{
 			testName:              "critical pods cannot be preempted",
+			isPodKillerWithError:  false,
 			inputPods:             []*v1.Pod{allPods[clusterCritical]},
 			insufficientResources: getAdmissionRequirementList(0, 0, 1),
 			expectErr:             true,
 			expectedOutput:        nil,
 		},
 		{
 			testName:              "best effort pods are not preempted when attempting to free resources",
+			isPodKillerWithError:  false,
 			inputPods:             []*v1.Pod{allPods[bestEffort]},
 			insufficientResources: getAdmissionRequirementList(0, 1, 0),
 			expectErr:             true,
 			expectedOutput:        nil,
 		},
 		{
-			testName: "multiple pods evicted",
+			testName:             "multiple pods evicted",
+			isPodKillerWithError: false,
 			inputPods: []*v1.Pod{
 				allPods[clusterCritical], allPods[bestEffort], allPods[burstable], allPods[highRequestBurstable],
 				allPods[guaranteed], allPods[highRequestGuaranteed]},
 			insufficientResources: getAdmissionRequirementList(0, 550, 0),
 			expectErr:             false,
 			expectedOutput:        []*v1.Pod{allPods[highRequestBurstable], allPods[highRequestGuaranteed]},
 		},
+		{
+			testName:             "multiple pods with eviction error",
+			isPodKillerWithError: true,
+			inputPods: []*v1.Pod{
+				allPods[clusterCritical], allPods[bestEffort], allPods[burstable], allPods[highRequestBurstable],
+				allPods[guaranteed], allPods[highRequestGuaranteed]},
+			insufficientResources: getAdmissionRequirementList(0, 550, 0),
+			expectErr:             false,
+			expectedOutput:        nil,
+		},
 	}
 	for _, r := range runs {
+		podProvider := newFakePodProvider()
+		podKiller := newFakePodKiller(r.isPodKillerWithError)
+		criticalPodAdmissionHandler := getTestCriticalPodAdmissionHandler(podProvider, podKiller)
```

Comment:
Previously these 3 items where created once and reused for all test cases. Now they're re-created for every test case. Is it done intentionlaly?

Reply:
Yeah, in the previous version, the input parameters in newFakePodKiller(r.isPodKillerWithError) were separated so they could be reused. This time, I combined these two because there was a lot of duplicate code. After merging them, I think it's better to recreate them every time during initialization.

Reply:
> newFakePodKiller(r.isPodKillerWithError) were separated so they could be reused

They can be reused in the current code as well if you create 2 podKiller objects and refer them in the test cases instead of the `isPodKillerWithError` flag.

Either way is ok for me. I was just curious if it was intentional.

Reply:
Thanks for your review! I think both methods are OK. In the test file of table-drivern, I think the flag method is simpler and clearer.